File: QadeerApp.Web/Modules/Administration/Language/LanguageRow.cs
namespace QadeerApp.Administration;

[ConnectionKey("Default"), Module("Administration"), TableName("Languages")]
[DisplayName("Languages"), InstanceName("Language")]
[ReadPermission(PermissionKeys.Language.View)]
[ModifyPermission(PermissionKeys.Language.Update)]
[InsertPermission(PermissionKeys.Language.Insert)]
[UpdatePermission(PermissionKeys.Language.Update)]
[DeletePermission(PermissionKeys.Language.Delete)]
[LookupScript(Permission = "*")]
public sealed class LanguageRow : Row<LanguageRow.RowFields>, IIdRow, INameRow
{
    [DisplayName("Language Code"), Size(10), NotNull, IdProperty, Unique]
    public string LanguageId { get => fields.LanguageId[this]; set => fields.LanguageId[this] = value; }

    [DisplayName("Language Name"), Size(50), NotNull, QuickSearch, NameProperty]
    public string LanguageName { get => fields.LanguageName[this]; set => fields.LanguageName[this] = value; }

    public class RowFields : RowFieldsBase
    {
        public StringField LanguageId;
        public StringField LanguageName;
    }
}

File: QadeerApp.Web/Modules/ServerTypes/Administration/LanguageRow.ts
import { fieldsProxy, getLookup, getLookupAsync } from "@serenity-is/corelib";

export interface LanguageRow {
    LanguageId?: string;
    LanguageName?: string;
}

export abstract class LanguageRow {
    static readonly idProperty = 'LanguageId';
    static readonly nameProperty = 'LanguageName';
    static readonly localTextPrefix = 'Administration.Language';
    static readonly lookupKey = 'Administration.Language';

    /** @deprecated use getLookupAsync instead */
    static getLookup() { return getLookup<LanguageRow>('Administration.Language') }
    static async getLookupAsync() { return getLookupAsync<LanguageRow>('Administration.Language') }

    static readonly deletePermission = 'Administration:Translation:Delete';
    static readonly insertPermission = 'Administration:Translation:Insert';
    static readonly readPermission = 'Administration:Translation:View';
    static readonly updatePermission = 'Administration:Translation:Update';

    static readonly Fields = fieldsProxy<LanguageRow>();
}
